python/cuml/cluster/kmeans.pyx

```diff
@@ -290,26 +291,34 @@ class KMeans(Base):
         cdef KMeansParams params
         params.n_clusters = <int>self.n_clusters
 
-        # K-means++ is the constrained case of k-means||
-        # w/ oversampling factor = 0
-        if (init == 'k-means++'):
-            init = 'k-means||'
-            self.oversampling_factor = 0
-
-        if (init in ['scalable-k-means++', 'k-means||']):
-            self.init = init
-            params.init = KMeansPlusPlus
-
-        elif (init == 'random'):
-            self.init = init
-            params.init = Random
-
-        else:
+        # cuPy does not allow comparing with string. See issue #2372
+        if isinstance(init, cupy.ndarray):
```

Comment:
Could we change this to reuse the same code for 'preset' that exists below? How about starting with:
```
init_str = init if isinstance(init, str) else None
...
if init_str == 'k-means++':
 init_str = 'kmeans||'
...
```
 Then we'd just need to do this chunk of code once?